src/core/js/models/itemsQuestionModel.js

```diff
@@ -0,0 +1,217 @@
+        setScore: function() {
+            var questionWeight = this.get('_questionWeight');
+            var answeredCorrectly = this.get('_isCorrect');
+            var score = answeredCorrectly ? questionWeight : 0;
+            this.set('_score', score);
+        },
+
+        setupFeedback: function() {
+            if (!this.has('_feedback')) return;
+
+            if (this.get('_isCorrect')) {
+                this.setupCorrectFeedback();
+                return;
+            }
+
+            if (this.isPartlyCorrect()) {
+                this.setupPartlyCorrectFeedback();
+                return;
+            }
+
+            // apply individual item feedback
+            var activeItem = this.getActiveItem();
+            if (this.isSingleSelect() && activeItem.get('feedback')) {
+                this.setupIndividualFeedback(activeItem);
+                return;
+            }
+
+            this.setupIncorrectFeedback();
+        },
+
+        setupIndividualFeedback: function(selectedItem) {
+            // for compatibility with framework v2
+            var title = this.getFeedbackTitle ?
+                this.getFeedbackTitle(this.get('_feedback')) :
+                this.get('title');
+
+            this.set({
+                feedbackTitle: title,
+                feedbackMessage: selectedItem.get("feedback")
+            });
+        },
+
+        isPartlyCorrect: function() {
+            return this.get('_isAtLeastOneCorrectSelection');
+        },
+
+        resetUserAnswer: function() {
+            this.set('_userAnswer', []);
+        },
+
+        isAtActiveLimit: function() {
+            var selectedItems = this.getActiveItems();
+            return (selectedItems.length === this.get('_selectable'));
+        },
+
+        isSingleSelect: function() {
+            return (this.get('_selectable') === 1);
+        },
+
+        getLastActiveItem: function(){
+            var selectedItems = this.getActiveItems();
+            return selectedItems[selectedItems.length-1];
+        },
+
+        resetItems: function() {
+            this.resetActiveItems();
+            this.set({
+                _isAtLeastOneCorrectSelection: false
+            });
+        },
+
+        getInteractionObject: function() {
+            var interactions = {
+                correctResponsesPattern: [],
+                choices: []
+            };
+
+            interactions.choices = this.getChildren().map(function(itemModel) {
+                return {
+                    id: (itemModel.get('_index') + 1).toString(),
+                    description: itemModel.get('text')
+                };
+            });
+
+            var correctItems = this.getChildren().filter(function(itemModel) {
+                return itemModel.get('_shouldBeSelected');
+            });
+
+            interactions.correctResponsesPattern = [
+                _.map(correctItems, function(itemModel) {
```

Comment:
Is it possible to use native `.map()` here?

Reply:
```
var correctItems = this.getChildren().filter
```
[Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) Always returns an Array. No need for typing here ;)

Reply:
Fair play. [https://underscorejs.org/#filter]
Most of those underscore functions on the Collection return their normal stuffs but use the Collection.models as the subject argument of the function instead of an Array.